rust/src/client/request.rs

```diff
@@ -269,14 +269,17 @@ impl Proxy {
 /// Manages client requests over HTTP.
 pub struct ClientReq {
     client: Client,
-    address: &'static str,
+    address: String,
 }
 
 impl ClientReq {
-    fn new(address: &'static str) -> Self {
+    fn new<S>(address: S) -> Self
+    where
+        S: Into<String>,
+    {
```

Comment:
In general, it is not bad practice but here we can't use `&'static str` because the `address` is not known at compile time.

I found a good explanation of `&'static str` on [rust-lang.org](https://users.rust-lang.org/t/quick-question-static-str/35940/2):

> As a conclusion, &'static str is a reference to the UTF-8 encoded variable length of byte sequence, which is valid for the entire lifetime of the process. But how can we obtain such type? Normally it's from the string literals, which are embeded directly on the executable binary(like .rodata section of the elf file) and loaded to the read-only section of the memory by the OS before execution.

Reply:
ah, right :+1: